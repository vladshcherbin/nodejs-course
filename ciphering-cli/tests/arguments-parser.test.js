import parseArguments from '../src/arguments-parser'

test('Return empty arguments object', () => {
  const args = []

  expect(parseArguments(args)).toStrictEqual({})
})

test('Return argument with no value', () => {
  const args = ['-c']

  expect(parseArguments(args)).toStrictEqual({
    config: true
  })
})

test('Return argument with value', () => {
  const args = ['-c', 'C1-R1']

  expect(parseArguments(args)).toStrictEqual({
    config: 'C1-R1'
  })
})

test('Return known arguments', () => {
  const args = ['-c', 'C1-A', '--input', 'input.txt', '-o', 'output.txt']

  expect(parseArguments(args)).toStrictEqual({
    config: 'C1-A',
    input: 'input.txt',
    output: 'output.txt'
  })
})

test('Skip unknown arguments', () => {
  const args = ['-c', 'C1-A', '-u', 'unknown']

  expect(parseArguments(args)).toStrictEqual({
    config: 'C1-A'
  })
})

test('Throw error on duplicate argument', () => {
  const args = ['-c', 'C1-A', '--config']

  expect(() => parseArguments(args)).toThrow('"config" option is duplicated')
})
