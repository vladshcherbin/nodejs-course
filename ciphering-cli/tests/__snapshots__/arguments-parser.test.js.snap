// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Throw error on duplicate argument 1`] = `"\\"config\\" option is duplicated"`;
